crates/runtime/src/accelerated_table.rs

```diff
@@ -60,38 +71,71 @@ impl AcceleratedTable {
         refresh_interval: Option<Duration>,
         object_store: Option<(Url, Arc<dyn ObjectStore + 'static>)>,
     ) -> Self {
+        let (refresh_trigger, refresh_trigger_receiver) = mpsc::channel::<()>(1);
```

Comment:
The [`tokio::sync::mspc::Receiver`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Receiver.html) struct is `Sync` + `Send` - so it should be movable between threads. Its true that it can't be copied - but that is part of the design (multiple producer, *single* consumer). So when constructing the enum you would need to move the value, not copy it.

Reply:
Confirmed that the plan is to use AcceleratedRefreshMode instance for refresh loop initialization, no need to keep it as part of main struct, so we have single instance.